marlowe-playground-client/src/Simulation.purs

```diff
@@ -132,16 +135,26 @@ handleAction _ (SetEditorText contents) = do
   editorSetValue contents
   updateContractInState contents
 
+handleAction _ (SetInitialSlot initialSlot) = do
+  assign (_currentMarloweState <<< _executionState <<< _SimulationNotStarted <<< _initialSlot) initialSlot
+
 handleAction _ StartSimulation = do
-  assign (_currentMarloweState <<< _executionState) (Just $ emptyExecutionStateWithSlot zero)
-  moveToSlot zero
+  maybeInitialSlot <- peruse (_currentMarloweState <<< _executionState <<< _SimulationNotStarted <<< _initialSlot)
+  case maybeInitialSlot of
+    Just initialSlot -> do
+      saveInitialState
+      assign (_currentMarloweState <<< _executionState) (emptyExecutionStateWithSlot initialSlot)
+      moveToSignificantSlot initialSlot
+      mCurrContract <- use _currentContract
+      case mCurrContract of
+        Just currContract -> editorSetValue (show $ genericPretty currContract)
+        Nothing -> pure unit
+    Nothing -> pure unit
 
 handleAction _ (MoveSlot slot) = do
-  maybeExecutionState <- use (_currentMarloweState <<< _executionState)
-  let
-    slotGTcurrentSlot = maybe false (\x -> slot > (x ^. _slot)) maybeExecutionState
+  inTheFuture <- hasable (_currentMarloweState <<< _executionState <<< _SimulationRunning <<< _slot <<< nearly zero ((>) slot))
```

Comment:
given that there is an `inFuture` function for this on line 692, I think it would be best to use that here, I think with
```purescript
inTheFuture <- inFuture <$> get
```

Reply:
It is out of scope right? It is a different function

Reply:
yeah but I mean you could move it to be a top-level function